### PR TITLE
Add New Job-Roles

### PR DESCRIPTION
**Pull request by Shreyangi**
Test engineer description (PLOW-135)

Co-authored-by: Arne Rusek <arne.rusek@mayadata.io>
Signed-off-by: Shreyangi <shreyangi.saxena@mayadata.io>

### CODE REVIEW

**Comment by Shreyangi** [comment]
@kmova @bobek  Can you please review and merge this?

**Comment by Shreyangi** [comment]
Can you please review this?

File: job-roles/principal-engineer.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- Capable of meeting tight deadlines with excellent overall project and product development life cycle experience and team spirit.
+- Experience in agile and scrum methodologies.
+- Linux OS internals experience.
+- Excellent knowledge of multithreading, thread-pools with strong OOP skills.
+
+## Nice-to-have:
+
+- Linux OS internals experience.
+- Expert level Rust/C/C++ coding and debugging; kernel, protocol, or hardware programming.
+- Experience in LINUX IDE.
```
Comment (bobek) [inline job-roles/principal-engineer.md:34]: Not sure, what is meant by LINUX IDE? Like, that the person should be able to use editor or some particular IDE? I would suggest removal of this item.

File: job-roles/principal-engineer.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- Mentoring software engineers, fostering an environment of trust and accountability.
+
+## Requirements:
+
+- Programming experience collaborating on software development Open Source Community.
+- Programming in C or C++ or Rust, or other systems programming languages.
+- Capable of meeting tight deadlines with excellent overall project and product development life cycle experience and team spirit.
+- Experience in agile and scrum methodologies.
+- Linux OS internals experience.
+- Excellent knowledge of multithreading, thread-pools with strong OOP skills.
```
Comment (bobek) [inline job-roles/principal-engineer.md:28]: Not sure, why are we explicitly mentioning strong OOP skills. I would argue that functional programming is more valuable for system engineering than OOP.

**Review by bobek — CHANGES_REQUESTED** [review]
Thank you for your PR. I have linked your documents to the TOC (`SUMMARY.md`) they would not get build or be accessible without it. I took a liberty to make them a bit more consistent with the other job-positions. Please take a look at the particular commit.

BTW "test engineer" is mentioned in the subject, but is not part of the PR.

File: job-roles/principal-engineer.md (hunk 1 above)
Comment (Shreyangi) [reply job-roles/principal-engineer.md]: Removed and updated the same 
Thank you

File: job-roles/principal-engineer.md (hunk 2 above)
Comment (Shreyangi) [reply job-roles/principal-engineer.md]: Thank you for the suggestion. I have added it

**Comment by Shreyangi** [comment]
I have updated the files with required changes

**Review by bobek — APPROVED** [review]

**Pull request merged**